### DISCUSSION TITLE
Which version to choose if CPU supports both AVX2 and SSE3?

### DISCUSSION THREAD

**Discussion by umitseyhan75** [Alex313031/thorium#241] [Q&A] [answered]
Windows 10 22H2, Intel Core i5-7200U.

**Comment by gz83** [comment]
You can use the AVX2 version

@umitseyhan75

**Reply by umitseyhan75** [reply]
Would using the AVX2 version proivide any benefit over regular Windows builds?

**Reply by gz83** [reply, marked as answer]
The regular windows version is the AVX version, which requires the processor to support the AVX instruction set for proper use, and the AVX2 version requires the processor to support the AVX2 instruction set for proper use.

The difference between the AVX and AVX2 instruction sets involves knowledge of operations, algorithms, etc. In general: AVX2 extends vector processing capabilities across floating point and integer data domains compared to AVX.

Due to the large differences in usage environment, different browsers, or different versions of code, the AVX or AVX2 versions usually have different performance, i.e., the AVX version sometimes performs better than the AVX2 version, and vice versa.

To summarize, you can choose the AVX or AVX2 version that performs better for you.

Some other references:

https://stackoverflow.com/questions/68340319/differences-between-avx-and-avx2

https://en.wikipedia.org/wiki/Advanced_Vector_Extensions

@umitseyhan75
